File: 34. Find First and Last Position of Element in Sorted Array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(arr, x) {
    let n=arr.length
    function start(arr,x){
        let l=0
        let h=n-1
        let ans=-1
        while (h>=l) {
            let mid=(l+h)//2
            if (arr[mid]==x){
                ans=mid
                h=mid-1
            }
            else if (arr[mid]<x) {
                l=mid+1
            }else{
                h=mid-1
            }
        }
        return ans
    }    
    function end(arr,x){
        let l=0
        let h=n-1
        let ans=-1
        while (h>=l) {
            let mid=(l+h)//2
            if (arr[mid]==x){
                ans=mid
                l=mid+1
            }
            else if (arr[mid]<x) {
                l=mid+1
            }else{
                h=mid-1
            }
        }return ans
    }
    return [start(arr,x), end(arr,x)]
            
    
};

File: 34.-Find-First-and-Last-Position-of-Element-in-Sorted-Array.py
class Solution:
    def searchRange(self, arr: List[int], x: int) -> List[int]:
        # code here
        n=len(arr)
        def start(arr,x):
            l=0
            h=n-1
            ans=-1
            while (h>=l) :
                mid=(l+h)//2
                if arr[mid]==x:
                    ans=mid
                    h=mid-1
                elif arr[mid]<x :
                    l=mid+1
                else:
                    h=mid-1
            return ans
            
        def end(arr,x):
            l=0
            h=n-1
            ans=-1
            while h>=l :
                mid=(l+h)//2
                if arr[mid]==x:
                    ans=mid
                    l=mid+1
                elif arr[mid]<x:
                    l=mid+1
                else:
                    h=mid-1
            return ans
        return [start(arr,x), end(arr,x)]
            
        
